functions/v2/firebase/hello-firestore/main_test.py

```diff
@@ -0,0 +1,51 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from unittest.mock import MagicMock
+
+from google.events.cloud import firestore
+
+from main import hello_firestore
+
+
+def test_hello_firestore(capsys):
+    old_document = firestore.Document(
+        name="/projects/test-project/databases/(default)/documents/test-collection/test-doc",
+        fields={"name": firestore.Value(string_value="Test Name")},
+    )
+    new_document = firestore.Document(
+        name="/projects/test-project/databases/(default)/documents/test-collection/test-doc",
+        fields={"name": firestore.Value(string_value="New Test Name")},
+    )
+
+    firestore_payload = firestore.DocumentEventData(
+        old_value=old_document,
+        value=new_document,
+    )
+
+    cloud_event = MagicMock()
```

Comment:
Instead of mocks, would it be possible to test using the actual product?

Doesn't have to be end-to-end, but we want to make sure that the code runs as expected over the actual product rather than testing this against the mock with just the client library.

Reply:
Changed it to use the correct CloudEvent class instead of mock. Thanks for pointing this out.